src/main/java/org/cloudfoundry/promregator/auth/OAuth2XSUAAEnricher.java

```diff
@@ -1,188 +1,87 @@
 package org.cloudfoundry.promregator.auth;
 
 import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
-import java.time.Instant;
-import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.Base64;
-import java.util.List;
 
-import org.apache.http.Consts;
-import org.apache.http.NameValuePair;
-import org.apache.http.ParseException;
-import org.apache.http.client.ClientProtocolException;
-import org.apache.http.client.config.RequestConfig;
-import org.apache.http.client.entity.UrlEncodedFormEntity;
-import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpPost;
 import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.message.BasicNameValuePair;
-import org.apache.http.util.EntityUtils;
 import org.cloudfoundry.promregator.config.OAuth2XSUAAAuthenticationConfiguration;
-
-import com.google.gson.Gson;
-import com.google.json.JsonSanitizer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.sap.cloud.security.client.HttpClientFactory;
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.xsuaa.client.DefaultOAuth2TokenService;
+import com.sap.cloud.security.xsuaa.client.XsuaaDefaultEndpoints;
+import com.sap.cloud.security.xsuaa.tokenflows.ClientCredentialsTokenFlow;
+import com.sap.cloud.security.xsuaa.tokenflows.TokenFlowException;
+import com.sap.cloud.security.xsuaa.tokenflows.XsuaaTokenFlows;
+
 public class OAuth2XSUAAEnricher implements AuthenticationEnricher {
-	
+
 	private static final Logger log = LoggerFactory.getLogger(OAuth2XSUAAEnricher.class);
-	
-	static final CloseableHttpClient httpclient = HttpClients.createDefault();
 
-	private final OAuth2XSUAAAuthenticationConfiguration config;
-	
-	public OAuth2XSUAAEnricher(OAuth2XSUAAAuthenticationConfiguration config) {
+	private final CloseableHttpClient httpClient;
+	private final ClientCredentialsTokenFlow tokenClient;
+
+	OAuth2XSUAAEnricher(OAuth2XSUAAAuthenticationConfiguration config) {
+		this(config, null);
+	}
+
+	OAuth2XSUAAEnricher(OAuth2XSUAAAuthenticationConfiguration config, ClientCredentialsTokenFlow tokenClient) {
 		super();
-		this.config = config;
+		OAuth2ServiceConfiguration c = new OAuth2ServiceConfig(config);
+
+		if (tokenClient != null) {
+			this.httpClient = null;
+			this.tokenClient = tokenClient;
+		} else {
+			this.httpClient = HttpClientFactory.create(c.getClientIdentity());
+			this.tokenClient = new XsuaaTokenFlows(new DefaultOAuth2TokenService(this.httpClient),
+					new XsuaaDefaultEndpoints(c), c.getClientIdentity()).clientCredentialsTokenFlow();
+		}
+		this.tokenClient.scopes(config.getScopes().toArray(new String[0]));
```

Comment:
Feels a little strange this case here, as if `tokenClient` is provided to this constructor, we would cause a side-effect here. 

BTW: What's your primary intent for providing `tokenClient` here? Is this "only for unit testing"?

Reply:
> Is this "only for unit testing"?

Yes, this is "only for unit testing". As always in life there are several options. I can adjust to the project habits ...

Reply:
It's okay. However, I would feel better, if we javadoc-commented that using this constructor here is discouraged due to potential side-effects.

Reply:
done